catalyst-gateway/bin/src/event_db/follower.rs

```diff
@@ -15,99 +16,167 @@ pub type BlockHash = String;
 /// Unique follower id
 pub type MachineId = String;
 
+/// `slot_no` column name
+const SLOT_NO_COLUMN: &str = "slot_no";
+/// `block_hash` column name
+const BLOCK_HASH_COLUMN: &str = "block_hash";
+/// `block_time` column name
+const BLOCK_TIME_COLUMN: &str = "block_time";
+/// `ended` column name
+const ENDED_COLUMN: &str = "ended";
+
+/// `insert_slot_index.sql`
+const INSERT_SLOT_INDEX_SQL: &str =
+    include_str!("../../../event-db/queries/follower/insert_slot_index.sql");
+/// `select_slot_info_by_datetime.sql.tmpl`
+const SLOT_INFO_SQL_TMPL: &str =
+    include_str!("../../../event-db/queries/follower/select_slot_info_by_datetime.sql.tmpl");
+
+/// Query type
+pub(crate) enum SlotInfoQueryType {
+    /// Previous slot info query type
+    Previous,
+    /// Current slot info query type
+    Current,
+    /// Next slot info query type
+    Next,
+}
+
+/// Slot info query template fields
+#[derive(serde::Serialize)]
+struct SlotInfoQueryTmplFields {
+    /// `sign` field from the sql template
+    sign: &'static str,
+    /// `ordering` field from the sql template
+    ordering: Option<&'static str>,
+}
+
+impl SlotInfoQueryType {
+    /// Get SQL query
+    fn get_sql_query(&self) -> Result<String, Error> {
+        let mut reg = Handlebars::new();
+        // disable default `html_escape` function
+        // which transforms `<`, `>` symbols to `&lt`, `&gt`
+        reg.register_escape_fn(|s| s.into());
+        let query = match self {
+            SlotInfoQueryType::Previous => {
```

Comment:
I wouldn't do this,  just set up the data using the match.

Have the render run by itself after you have set up the data.
Something like this:
```
let data = match self {
  SlotInfoQueryType::Previous => SlotInfoQueryFields{
    blah
  },
 and so on
}

reg.render_template(SLOT_INFO_SQL_TMPL, &data)
```